s3-trigger-stack.yaml

```diff
@@ -79,9 +79,32 @@ Resources:
               response = {}
               try:
                   if event['RequestType'] == 'Delete':
-                      cfnresponse.send(event, context, cfnresponse.SUCCESS, {})
-                      return {}
-                  
+                      try:
+
+                          event_data = event['ResourceProperties']
+                          lambda_arn = event_data.get('LambdaFunctionArn', '')
+                          buckets = event_data.get('S3BucketArns', [])
+                          for bucket_arn in buckets:
+                              parts = bucket_arn.split(":::")[-1].split("/", 1)
+                              bucket_name = parts[0]
+                           
+                              # get the existing notification configuration
+                              existing_config = s3.get_bucket_notification_configuration(Bucket=bucket_name)
+
+                              existing_lambda_configs = existing_config.get('LambdaFunctionConfigurations', [])
+                              updated_lambda_configs = [config for config in existing_lambda_configs if config['LambdaFunctionArn'] != lambda_arn]
+ 
+                              notification_config = {
```

Comment:
Do we need to check prefix also?

Reply:
It is working for prefix as well. Here is how I tested:
1. Create a bucket B1. and then create two folders (prefix) inside P1 and P2.
2. Go and manually add trigger for P1 and point it to some lambda xyz.
3. Now create a stack and give bucket B1 and prefix P2.
4. You can see two triggers for the bucket.
5. Delete the stack, only the trigger for P2 will be deleted since this will be associated with our lambda.

Let me know, if you want some other test.